ambari-server/src/main/resources/stacks/BIGTOP/3.2.0/services/SPARK/service_advisor.py

```diff
@@ -163,115 +163,100 @@ def isKerberosEnabled(services, configurations):
-
-        putSpark2AtlasHookProperty("atlas.client.checkModelInStart", "false")
+    if "spark-atlas-application-properties-override" in services["configurations"]:
+      spark_atlas_application_properties_override = self.getServicesSiteProperties(services, "spark-atlas-application-properties-override")
+      spark_defaults_properties = self.getServicesSiteProperties(services, "spark-defaults")
+      putSparkDefautlsProperty = self.putProperty(configurations, "spark-defaults", services)
+      putSparkDefaultsPropertyAttribute = self.putPropertyAttribute(configurations,"spark-defaults")
+      putSparkAtlasHookProperty = self.putProperty(configurations, "spark-atlas-application-properties-override", services)
+      putSparkAtlasHookPropertyAttribute = self.putPropertyAttribute(configurations,"spark-atlas-application-properties-override")
+      spark_sac_enabled = None
+      if self.checkSiteProperties(spark_atlas_application_properties_override, "atlas.spark.enabled"):
+        spark_sac_enabled = spark_atlas_application_properties_override["atlas.spark.enabled"]
+        spark_sac_enabled = str(spark_sac_enabled).upper() == 'TRUE'
+
+      if spark_sac_enabled:
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.driver.extraClassPath", "/usr/bigtop/current/spark-atlas-connector/*", ":")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.yarn.dist.files", "/etc/spark/conf/atlas-application.properties.yarn#atlas-application.properties", ",")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.driver.extraClassPath", "/usr/bigtop/current/spark-atlas-connector/*", ":")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.extraListeners", "com.hortonworks.spark.atlas.SparkAtlasEventTracker", ",")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.sql.queryExecutionListeners", "com.hortonworks.spark.atlas.SparkAtlasEventTracker", ",")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.extraListeners", "com.hortonworks.spark.atlas.SparkAtlasEventTracker", ",")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.sql.queryExecutionListeners", "com.hortonworks.spark.atlas.SparkAtlasEventTracker", ",")
+
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.sql.streaming.streamingQueryListeners", "com.hortonworks.spark.atlas.SparkAtlasStreamingQueryEventTracker", ",")
+        self.setOrAddValueToProperty(putSparkDefautlsProperty, spark_defaults_properties, "spark.sql.streaming.streamingQueryListeners", "com.hortonworks.spark.atlas.SparkAtlasStreamingQueryEventTracker", ",")
```

Comment:
`com.hortonworks.spark.atlas.SparkAtlasStreamingQueryEventTracker com.hortonworks.spark.atlas.SparkAtlasEventTracker
`
Should we keep these classes here in bigtop stack?

Reply:
We can remove it for now since we haven't integrated Atlas yet. The function "recommendSPARKConfigurationsFromHDP30" is mainly used to set properties related to Spark SAC and Atlas. Therefore, we can remove this function first and add the testing advisor related to it later when we have integrated the relevant services.